ci/continuous_integration/daemon/main.py

```
@@ -78,16 +77,73 @@ def update_auth_header():
   os.chmod(AUTH_FILE_LOCATION, 0600)
 
 
+def get_version():
+  """Returns the version of the binary."""
+
+  proc = subprocess.Popen(build_command('supported_job_types'), stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+  out, _ = proc.communicate()
+  return yaml.load(out)['Version']
+
+
+def get_supported_jobtypes():
+  """Returns a hash of supported job types."""
+
+  proc = subprocess.Popen(build_command('supported_job_types'), stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True)
+  out, _ = proc.communicate()
+  result = yaml.load(out)
+  result.pop('Version', None)
+  return result
+
+
+def load_new_testcases(latest_testcase=None):
+  """Returns a new list of testcases from clusterfuzz to run."""
+  with open(AUTH_FILE_LOCATION, 'r') as f:
+    auth_header = f.read()
+
+  testcases = None
+  page = 1
+  supported_jobtypes = get_supported_jobtypes()
+  while not testcases:
+    r = requests.post('https://clusterfuzz.com/v2/testcases/load',
+                      headers={'Authorization': auth_header},
+                      json={'page': page,
+                            'reproducible': 'yes'})
+    testcases = json.loads(r.text)['items']
+    testcases = [testcase['id'] for testcase in testcases
+                 if testcase['jobType'] in supported_jobtypes['chromium']]
+    if latest_testcase in testcases:
+      testcases = testcases[0:testcases.index(latest_testcase)]
+    page = random.randint(5, 35)
+  return testcases
+
+
+def reset_and_run_testcase(testcase_id, test_type):
+  """Resets the chromium repo and runs the testcase."""
+
+  if os.path.exists(CHROMIUM_OUT):
+    shutil.rmtree(CHROMIUM_OUT)
+  if os.path.exists(CLUSTERFUZZ_DIR):
+    shutil.rmtree(CLUSTERFUZZ_DIR)
+  subprocess.call('git checkout -f master', shell=True, cwd=CHROMIUM_SRC)
```

This one fails silently. Either use `check_call` or refactor it to `common.execute`.